common/lib/xmodule/xmodule/assets/vertical/public/js/vertical_student_view.js

```diff
@@ -24,32 +25,43 @@ window.VerticalStudentView = function(runtime, element) {
             apiUrl: $bookmarkButtonElement.data('bookmarksApiUrl')
         });
     });
-    $(element).find('.vert').each(
-        function(idx, block) {
-            var blockKey = block.dataset.id;
-
-            if (block.dataset.completableByViewing === undefined) {
-                return;
-            }
-            // TODO: EDUCATOR-1778
-            // *  Check if blocks are in the browser's view window or in focus
-            //    before marking complete. This will include a configurable
-            //    delay so that blocks must be seen for a few seconds before
-            //    being marked complete, to prevent completion via rapid
-            //    scrolling.  (OC-3358)
-            // *  Limit network traffic by batching and throttling calls.
-            //    (OC-3090)
-            if (blockKey && !SEEN_COMPLETABLES.has(blockKey)) {
-                $.ajax({
-                    type: 'POST',
-                    url: runtime.handlerUrl(element, 'publish_completion'),
-                    data: JSON.stringify({
-                        block_key: blockKey,
-                        completion: 1.0
-                    })
-                });
-                SEEN_COMPLETABLES.add(blockKey);
+    RequireJS.require(['bundles/ViewedEvent'], function(ViewedEvent) {
+        var tracker;
+        var completableBlocks = [];
+        var viewedAfter = element.dataset.viewedAfter || 5000;
+        $(element).find('.vert').each(function(idx, block) {
+            if (block.dataset.completableByViewing !== undefined) {
+                completableBlocks.push(block);
             }
+        });
+        if (completableBlocks) {
+            tracker = new ViewedEventTracker(completableBlocks, viewedAfter);
```

Comment:
For some reason, the `ViewedEvent` variable on L28 above is undefined, but the import works, and the name `ViewedEventTracker` is available.  I thought this would import the module as ViewedEvent, and I'd have to access the object as `ViewedEvent.ViewedEventTracker`.  Clearly, I don't understand how RequireJS works.

Reply:
Hmm, that definitely looks wrong to me. But it appears that edx-platform has intentionally configured WebPack to add all exported symbols from each bundle to the global namespace (`window`) when the bundle is loaded. So it appears that RequireJS isn't loading a UMD/AMD module here, but is just doing a dumb import of a JS file and executing it, which has a side effect of adding symbols to the global namespace.

Reply:
@andy-armstrong may have some thoughts on the right way to do this.  (or might be able to point us in the right direction).

Andy, are you still in charge of FedX?

Reply:
@arizzitano Would you be able to tell us what's going on here, and what we should be doing? I feel like secret global imports are not the way. ^